### ISSUE TITLE
Blank page on login to HTTPS

### ISSUE THREAD

**Issue by pattertj** [linuxserver/docker-code-server#81]
[linuxserverurl]: https://linuxserver.io
[![linuxserver.io](https://raw.githubusercontent.com/linuxserver/docker-templates/master/linuxserver.io/img/linuxserver_medium.png)][linuxserverurl]

------------------------------

## Expected Behavior
Code-Server should load up the IDE

## Current Behavior
After entering my password, it loads a blank page.

## Steps to Reproduce

1. Build code-server container using Docker Compose
2. Configure Router Port Forwarding
3. Configure Nginx
4. Configure Let's Encrypt SSL
4. Access https://my.domain.com
5. Enter password

Accessing it via internal IP and HTTP works fine
Accessing it via the above steps through the proxy and HTTPS fails

## Environment
**OS:**
Portainer running on Proxmox
**CPU architecture:** 
Ryzen 5 2600x
**How docker service was installed:**

Docker compose

## Command used to create docker container (run/create/compose/screenshot)

```
---
version: "2.1"
services:
  code-server:
    image: ghcr.io/linuxserver/code-server
    container_name: code-server
    environment:
      - PUID=${PUID} # default user id, defined in .env
      - PGID=${PGID} # default group id, defined in .env
      - TZ=${TZ} # timezone, defined in .env
      - PASSWORD=password
      - SUDO_PASSWORD=sudopassword
      - PROXY_DOMAIN=my.domain.com
    volumes:
      - ${ROOT}/config/codeserver:/config
    ports:
      - 8443:8443
    restart: unless-stopped
```

## Docker logs

```
-------------------------------------

          _         ()

         | |  ___   _    __

         | | / __| | |  /  \ 

         | | \__ \ | | | () |

         |_| |___/ |_|  \__/



Brought to you by linuxserver.io

-------------------------------------


To support LSIO projects visit:

https://www.linuxserver.io/donate/

-------------------------------------

GID/UID

-------------------------------------


User uid:    1000

User gid:    1000

-------------------------------------


[cont-init.d] 10-adduser: exited 0.

[cont-init.d] 30-config: executing... 

setting up sudo access

setting sudo password using SUDO_PASSWORD env var

Enter new UNIX password: Retype new UNIX password: passwd: password updated successfully

[cont-init.d] 30-config: exited 0.

[cont-init.d] 90-custom-folders: executing... 

[cont-init.d] 90-custom-folders: exited 0.

[cont-init.d] 99-custom-scripts: executing... 

[custom-init] no custom files found exiting...

[cont-init.d] 99-custom-scripts: exited 0.

[cont-init.d] done.

[services.d] starting services

[services.d] done.

[2021-09-16T15:58:00.944Z] info  code-server 3.11.1 acb5de66b71a3f8f1cc065df4633ecd3f9788d46

[2021-09-16T15:58:00.945Z] info  Using user-data-dir ~/data

[2021-09-16T15:58:00.958Z] info  Using config file ~/.config/code-server/config.yaml

[2021-09-16T15:58:00.958Z] info  HTTP server listening on http://0.0.0.0:8443 

[2021-09-16T15:58:00.958Z] info    - Authentication is enabled

[2021-09-16T15:58:00.958Z] info      - Using password from $PASSWORD

[2021-09-16T15:58:00.958Z] info    - Not serving HTTPS 

[2021-09-16T15:58:00.958Z] info    - Proxying the following domain:

[2021-09-16T15:58:00.959Z] info      - *.my.domain.com
```

Additionally I have developer console errors in HTTPS that do not appear in HTTP mode:

```
vscode.browserified.js:725 [vscode] failed to initialize VS Code
4.../../common/util @ vscode.browserified.js:725
o @ vscode.browserified.js:1
r @ vscode.browserified.js:1
(anonymous) @ vscode.browserified.js:1
vscode.browserified.js:726 Error: [vscode] Could not set body background to theme background color. Could not find colorThemeData in localStorage.
    at setBodyBackgroundToThemeBackgroundColor (vscode.browserified.js:667)
    at main (vscode.browserified.js:718)
    at Object.4.../../common/util (vscode.browserified.js:722)
    at o (vscode.browserified.js:1)
    at r (vscode.browserified.js:1)
    at vscode.browserified.js:1
4.../../common/util @ vscode.browserified.js:726
o @ vscode.browserified.js:1
r @ vscode.browserified.js:1
(anonymous) @ vscode.browserified.js:1
vscode.browserified.js:129 info  [Service Worker] registered
browserSocketFactory.ts:162 WebSocket connection to 'wss://code-server.tylerpatterson.me/?type=Management&reconnectionToken=e1c1da98-e843-458a-a363-ef6117a26b0a&reconnection=false&skipWebSocketFrames=false' failed: 
create @ browserSocketFactory.ts:162
connect @ browserSocketFactory.ts:212
o @ remoteAgentConnection.ts:191
s @ remoteAgentConnection.ts:234
a @ remoteAgentConnection.ts:311
i @ remoteAgentConnection.ts:335
d @ remoteAgentConnection.ts:408
VM93:1 Uncaught (in promise) SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
    at new ai (editorOverrideService.ts:67)
    at f._createInstance (instantiationService.ts:110)
    at f._createServiceInstance (instantiationService.ts:230)
    at f._createServiceInstanceWithOwner (instantiationService.ts:219)
    at f._createAndCacheServiceInstance (instantiationService.ts:208)
    at f._safeCreateAndCacheServiceInstance (instantiationService.ts:151)
    at f._getOrCreateServiceInstance (instantiationService.ts:135)
    at Object.get (instantiationService.ts:54)
    at L.initLayout (layout.ts:260)
ai @ editorOverrideService.ts:67
_createInstance @ instantiationService.ts:110
_createServiceInstance @ instantiationService.ts:230
_createServiceInstanceWithOwner @ instantiationService.ts:219
_createAndCacheServiceInstance @ instantiationService.ts:208
_safeCreateAndCacheServiceInstance @ instantiationService.ts:151
_getOrCreateServiceInstance @ instantiationService.ts:135
get @ instantiationService.ts:54
initLayout @ layout.ts:260
(anonymous) @ workbench.ts:141
invokeFunction @ instantiationService.ts:61
startup @ workbench.ts:134
open @ web.main.ts:98
Promise.then (async)
m @ workbench.web.api.ts:526
(anonymous) @ workbench.ts:503
(anonymous) @ workbench.ts:510
r._invokeFactory @ loader.js:1136
r.complete @ loader.js:1146
r._onModuleComplete @ loader.js:1772
r._onModuleComplete @ loader.js:1784
r._resolve @ loader.js:1732
r.defineModule @ loader.js:1375
o @ loader.js:1660
v @ loader.js:2167
r._invokeFactory @ loader.js:1136
r.complete @ loader.js:1146
r._onModuleComplete @ loader.js:1772
r._onModuleComplete @ loader.js:1784
r._resolve @ loader.js:1732
r.defineModule @ loader.js:1375
p @ loader.js:1822
(anonymous) @ workbench.nls.js:7
errors.ts:28 Uncaught Error: WebSocket close with status code 1006

Error: WebSocket close with status code 1006
    at WebSocket.<anonymous> (browserSocketFactory.ts:129)
    at errors.ts:28
(anonymous) @ errors.ts:28
setTimeout (async)
unexpectedErrorHandler @ errors.ts:26
onUnexpectedError @ errors.ts:63
O @ errors.ts:82
createStorageService @ web.main.ts:345
async function (async)
createStorageService @ web.main.ts:341
initServices @ web.main.ts:207
async function (async)
initServices @ web.main.ts:188
open @ web.main.ts:89
oe @ web.main.ts:390
m @ workbench.web.api.ts:526
(anonymous) @ workbench.ts:503
(anonymous) @ workbench.ts:510
r._invokeFactory @ loader.js:1136
r.complete @ loader.js:1146
r._onModuleComplete @ loader.js:1772
r._onModuleComplete @ loader.js:1784
r._resolve @ loader.js:1732
r.defineModule @ loader.js:1375
o @ loader.js:1660
v @ loader.js:2167
r._invokeFactory @ loader.js:1136
r.complete @ loader.js:1146
r._onModuleComplete @ loader.js:1772
r._onModuleComplete @ loader.js:1784
r._resolve @ loader.js:1732
r.defineModule @ loader.js:1375
p @ loader.js:1822
(anonymous) @ workbench.nls.js:7
VM94:1 Uncaught SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
    at new ai (editorOverrideService.ts:67)
    at f._createInstance (instantiationService.ts:110)
    at f._createServiceInstance (instantiationService.ts:230)
    at f._createServiceInstanceWithOwner (instantiationService.ts:219)
    at f._createAndCacheServiceInstance (instantiationService.ts:208)
    at f._safeCreateAndCacheServiceInstance (instantiationService.ts:151)
    at f._getOrCreateServiceInstance (instantiationService.ts:135)
    at f._createInstance (instantiationService.ts:88)
    at f.createInstance (instantiationService.ts:76)
ai @ editorOverrideService.ts:67
_createInstance @ instantiationService.ts:110
_createServiceInstance @ instantiationService.ts:230
_createServiceInstanceWithOwner @ instantiationService.ts:219
_createAndCacheServiceInstance @ instantiationService.ts:208
_safeCreateAndCacheServiceInstance @ instantiationService.ts:151
_getOrCreateServiceInstance @ instantiationService.ts:135
_createInstance @ instantiationService.ts:88
createInstance @ instantiationService.ts:76
acquireInstantiationService @ configurationService.ts:449
(anonymous) @ workbench.ts:207
setTimeout (async)
(anonymous) @ workbench.ts:206
invokeFunction @ instantiationService.ts:61
initServices @ workbench.ts:200
startup @ workbench.ts:132
open @ web.main.ts:98
async function (async)
open @ web.main.ts:89
oe @ web.main.ts:390
m @ workbench.web.api.ts:526
(anonymous) @ workbench.ts:503
(anonymous) @ workbench.ts:510
r._invokeFactory @ loader.js:1136
r.complete @ loader.js:1146
r._onModuleComplete @ loader.js:1772
r._onModuleComplete @ loader.js:1784
r._resolve @ loader.js:1732
r.defineModule @ loader.js:1375
o @ loader.js:1660
v @ loader.js:2167
r._invokeFactory @ loader.js:1136
r.complete @ loader.js:1146
r._onModuleComplete @ loader.js:1772
r._onModuleComplete @ loader.js:1784
r._resolve @ loader.js:1732
r.defineModule @ loader.js:1375
p @ loader.js:1822
(anonymous) @ workbench.nls.js:7
browserSocketFactory.ts:162 WebSocket connection to 'wss://code-server.tylerpatterson.me/?type=ExtensionHost&reconnectionToken=e68ee6af-7218-4e91-99fc-ef01b07f6b86&reconnection=false&skipWebSocketFrames=false' failed: 
create @ browserSocketFactory.ts:162
connect @ browserSocketFactory.ts:212
o @ remoteAgentConnection.ts:191
s @ remoteAgentConnection.ts:234
a @ remoteAgentConnection.ts:311
g @ remoteAgentConnection.ts:353
b @ remoteAgentConnection.ts:422
extensionHostManager.ts:75 Error received from starting extension host (kind: 2)
(anonymous) @ extensionHostManager.ts:75
Promise.then (async)
Zn @ extensionHostManager.ts:69
_createInstance @ instantiationService.ts:110
createInstance @ instantiationService.ts:76
(anonymous) @ abstractExtensionService.ts:602
_startExtensionHosts @ abstractExtensionService.ts:601
_initialize @ abstractExtensionService.ts:505
(anonymous) @ extensionService.ts:78
async function (async)
(anonymous) @ extensionService.ts:77
Promise.then (async)
Ma @ extensionService.ts:76
_createInstance @ instantiationService.ts:110
_createServiceInstance @ instantiationService.ts:230
_createServiceInstanceWithOwner @ instantiationService.ts:219
_createAndCacheServiceInstance @ instantiationService.ts:208
_safeCreateAndCacheServiceInstance @ instantiationService.ts:151
_getOrCreateServiceInstance @ instantiationService.ts:135
get @ instantiationService.ts:54
initLayout @ layout.ts:255
(anonymous) @ workbench.ts:141
invokeFunction @ instantiationService.ts:61
startup @ workbench.ts:134
open @ web.main.ts:98
async function (async)
open @ web.main.ts:89
oe @ web.main.ts:390
m @ workbench.web.api.ts:526
(anonymous) @ workbench.ts:503
(anonymous) @ workbench.ts:510
r._invokeFactory @ loader.js:1136
r.complete @ loader.js:1146
r._onModuleComplete @ loader.js:1772
r._onModuleComplete @ loader.js:1784
r._resolve @ loader.js:1732
r.defineModule @ loader.js:1375
o @ loader.js:1660
v @ loader.js:2167
r._invokeFactory @ loader.js:1136
r.complete @ loader.js:1146
r._onModuleComplete @ loader.js:1772
r._onModuleComplete @ loader.js:1784
r._resolve @ loader.js:1732
r.defineModule @ loader.js:1375
p @ loader.js:1822
(anonymous) @ workbench.nls.js:7
extensionHostManager.ts:76 Error: WebSocket close with status code 1006
    at WebSocket.<anonymous> (browserSocketFactory.ts:129)
(anonymous) @ extensionHostManager.ts:76
Promise.then (async)
Zn @ extensionHostManager.ts:69
_createInstance @ instantiationService.ts:110
createInstance @ instantiationService.ts:76
(anonymous) @ abstractExtensionService.ts:602
_startExtensionHosts @ abstractExtensionService.ts:601
_initialize @ abstractExtensionService.ts:505
(anonymous) @ extensionService.ts:78
async function (async)
(anonymous) @ extensionService.ts:77
Promise.then (async)
Ma @ extensionService.ts:76
_createInstance @ instantiationService.ts:110
_createServiceInstance @ instantiationService.ts:230
_createServiceInstanceWithOwner @ instantiationService.ts:219
_createAndCacheServiceInstance @ instantiationService.ts:208
_safeCreateAndCacheServiceInstance @ instantiationService.ts:151
_getOrCreateServiceInstance @ instantiationService.ts:135
get @ instantiationService.ts:54
initLayout @ layout.ts:255
(anonymous) @ workbench.ts:141
invokeFunction @ instantiationService.ts:61
startup @ workbench.ts:134
open @ web.main.ts:98
async function (async)
open @ web.main.ts:89
oe @ web.main.ts:390
m @ workbench.web.api.ts:526
(anonymous) @ workbench.ts:503
(anonymous) @ workbench.ts:510
r._invokeFactory @ loader.js:1136
r.complete @ loader.js:1146
r._onModuleComplete @ loader.js:1772
r._onModuleComplete @ loader.js:1784
r._resolve @ loader.js:1732
r.defineModule @ loader.js:1375
o @ loader.js:1660
v @ loader.js:2167
r._invokeFactory @ loader.js:1136
r.complete @ loader.js:1146
r._onModuleComplete @ loader.js:1772
r._onModuleComplete @ loader.js:1784
r._resolve @ loader.js:1732
r.defineModule @ loader.js:1375
p @ loader.js:1822
(anonymous) @ workbench.nls.js:7
VM97:1 Uncaught (in promise) SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
    at new ai (editorOverrideService.ts:67)
    at f._createInstance (instantiationService.ts:110)
    at f._createServiceInstance (instantiationService.ts:230)
    at f._createServiceInstanceWithOwner (instantiationService.ts:219)
    at f._createAndCacheServiceInstance (instantiationService.ts:208)
    at f._safeCreateAndCacheServiceInstance (instantiationService.ts:151)
    at f._getOrCreateServiceInstance (instantiationService.ts:135)
    at f._createInstance (instantiationService.ts:88)
    at f.createInstance (instantiationService.ts:76)
ai @ editorOverrideService.ts:67
_createInstance @ instantiationService.ts:110
_createServiceInstance @ instantiationService.ts:230
_createServiceInstanceWithOwner @ instantiationService.ts:219
_createAndCacheServiceInstance @ instantiationService.ts:208
_safeCreateAndCacheServiceInstance @ instantiationService.ts:151
_getOrCreateServiceInstance @ instantiationService.ts:135
_createInstance @ instantiationService.ts:88
createInstance @ instantiationService.ts:76
_createExtensionHostCustomers @ extensionHostManager.ts:202
(anonymous) @ extensionHostManager.ts:72
Promise.then (async)
Zn @ extensionHostManager.ts:80
_createInstance @ instantiationService.ts:110
createInstance @ instantiationService.ts:76
(anonymous) @ abstractExtensionService.ts:602
_startExtensionHosts @ abstractExtensionService.ts:601
_initialize @ abstractExtensionService.ts:505
(anonymous) @ extensionService.ts:78
async function (async)
(anonymous) @ extensionService.ts:77
Promise.then (async)
Ma @ extensionService.ts:76
_createInstance @ instantiationService.ts:110
_createServiceInstance @ instantiationService.ts:230
_createServiceInstanceWithOwner @ instantiationService.ts:219
_createAndCacheServiceInstance @ instantiationService.ts:208
_safeCreateAndCacheServiceInstance @ instantiationService.ts:151
_getOrCreateServiceInstance @ instantiationService.ts:135
get @ instantiationService.ts:54
initLayout @ layout.ts:255
(anonymous) @ workbench.ts:141
invokeFunction @ instantiationService.ts:61
startup @ workbench.ts:134
open @ web.main.ts:98
async function (async)
open @ web.main.ts:89
oe @ web.main.ts:390
m @ workbench.web.api.ts:526
(anonymous) @ workbench.ts:503
(anonymous) @ workbench.ts:510
r._invokeFactory @ loader.js:1136
r.complete @ loader.js:1146
r._onModuleComplete @ loader.js:1772
r._onModuleComplete @ loader.js:1784
r._resolve @ loader.js:1732
r.defineModule @ loader.js:1375
o @ loader.js:1660
v @ loader.js:2167
r._invokeFactory @ loader.js:1136
r.complete @ loader.js:1146
r._onModuleComplete @ loader.js:1772
r._onModuleComplete @ loader.js:1784
r._resolve @ loader.js:1732
r.defineModule @ loader.js:1375
p @ loader.js:1822
(anonymous) @ workbench.nls.js:7
15VM97:1 Uncaught (in promise) SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
    at new ai (editorOverrideService.ts:67)
    at f._createInstance (instantiationService.ts:110)
    at f._createServiceInstance (instantiationService.ts:230)
    at f._createServiceInstanceWithOwner (instantiationService.ts:219)
    at f._createAndCacheServiceInstance (instantiationService.ts:208)
    at f._safeCreateAndCacheServiceInstance (instantiationService.ts:151)
    at f._getOrCreateServiceInstance (instantiationService.ts:135)
    at f._createInstance (instantiationService.ts:88)
    at f.createInstance (instantiationService.ts:76)
ai @ editorOverrideService.ts:67
_createInstance @ instantiationService.ts:110
_createServiceInstance @ instantiationService.ts:230
_createServiceInstanceWithOwner @ instantiationService.ts:219
_createAndCacheServiceInstance @ instantiationService.ts:208
_safeCreateAndCacheServiceInstance @ instantiationService.ts:151
_getOrCreateServiceInstance @ instantiationService.ts:135
_createInstance @ instantiationService.ts:88
createInstance @ instantiationService.ts:76
_createExtensionHostCustomers @ extensionHostManager.ts:202
(anonymous) @ extensionHostManager.ts:72
../../../../../vs/workbench/services/extensions/worker/extensionHostWorker.js:72 Uncaught (in promise) Error: Unknown actor MainThreadWindow
    at s._doInvokeHandler (workbench.web.api.js:1827)
    at s._invokeHandler (workbench.web.api.js:1827)
    at s._receiveRequest (workbench.web.api.js:1827)
    at s._receiveOneMessage (workbench.web.api.js:1827)
    at workbench.web.api.js:1827
    at n.fire (workbench.web.api.js:60)
    at MessagePort.D.onmessage (workbench.web.api.js:1867)
```

**Comment by Roxedus** [comment]
Which browser and reverse proxy are you using?

**Comment by pattertj** [comment]
Microsoft Edge
Version 93.0.961.47 (Official build) (64-bit)

and 

Nginx

**Comment by Roxedus** [comment]
Post your reverse proxy for code-server

**Comment by pattertj** [comment]
I am using jc21/nginx-proxy-manager in Docker. It is GUI configured. Is there a particular portion I can screenshot for you?

**Comment by Roxedus** [comment]
do all the tabs for the vhost

**Comment by pattertj** [comment]
I did some more digging this morning into the console logs, I was initially stuck on the local storage failure at the beginning, but I noticed some WebSocket errors. Enabling WebSockets in Nginx resolved the issue. I still have the local storage console error, but code-server seems to be working okay and I can't see any issues working in it now.

![image](https://user-images.githubusercontent.com/20607476/133751061-9a64ba8c-211f-4fdc-9614-a0ec73b0b5f4.png)

Sorry to bother you with this issue, but I wanted to share the fix! Thanks for your support on this project.
